stdlib/LinearAlgebra/src/cholesky.jl

```diff
@@ -155,10 +155,14 @@ function _chol!(A::AbstractMatrix, ::Type{UpperTriangular})
     n = checksquare(A)
     @inbounds begin
         for k = 1:n
+            Akk = A[k,k]
+            if Akk isa Complex && Akk != Akk'
+                Akk = real(Akk)
+            end
```

Comment:
This is not type-stable (`Akk` can change type).  Better to just write:
```jl
Akk = real(A[k,k])
```
In the case of real elements, the `real` function is the identity and will be optimized out, and just taking the real part will be faster than checking `Akk != Akk'` in the complex case anyway.

Reply:
I agree with respect to `Akk != Akk´` and will remove this check.
The `Akk isa Complex` is necessary, because the element type of `A` is not restricted to `Number`s, but can be for example an `AbstractArray` or some other type, which does not have the `real` method.

Reply:
You can use
```
Akk = eltype(A) <: Number ? real(A[k,k]) : A[k,k]
```
I guess.